Change secret keys docs to recommend registering an interceptor (#841)
smallrye/smallrye-config: documentation/src/main/docs/config/secret-keys.md

secrets. However, there is still the basic problem that passwords and secrets are generally encoded simply as strings. 
Secret Keys provides a way to "lock" the configuration so that secrets do not appear unless explicitly enabled.

> To mark specific keys as secret, register an instance of `io.smallrye.config.SecretKeysConfigSourceInterceptor`

Reviewer: ```suggestion
To mark specific keys as secrets, register an instance of `io.smallrye.config.SecretKeysConfigSourceInterceptor`
```

using your own interceptor factory, like this:

```java
public class SecretKeysConfigInterceptorFactory implements ConfigSourceInterceptorFactory {
    @Override
    public ConfigSourceInterceptor getInterceptor(ConfigSourceInterceptorContext context) {
        return new SecretKeysConfigSourceInterceptor(Set.of("secret"));
    }
}
```

> You will need to register this factory to be found at runtime by creating a `META-INF/services/io.smallrye.config.ConfigSourceInterceptorFactory`

Reviewer 2: ```suggestion
Register this factory so that it can be found at runtime by creating a `META-INF/services/io.smallrye.config.ConfigSourceInterceptorFactory` file that contains the fully qualified name of this factory class.
```

> file containing the fully qualified name of this factory class.

Reviewer 2: ```suggestion
```

secrets. However, there is still the basic problem that passwords and secrets are generally encoded simply as strings. 
Secret Keys provides a way to "lock" the configuration so that secrets do not appear unless explicitly enabled.

> To mark specific keys as secret, register an instance of `io.smallrye.config.SecretKeysConfigSourceInterceptor` by using your interceptor factory as follows:

Reviewer: ```suggestion
To mark specific keys as secrets, register an instance of `io.smallrye.config.SecretKeysConfigSourceInterceptor` by using your interceptor factory as follows:
```

```java
public class SecretKeysConfigInterceptorFactory implements ConfigSourceInterceptorFactory {
    @Override
    public ConfigSourceInterceptor getInterceptor(ConfigSourceInterceptorContext context) {
        return new SecretKeysConfigSourceInterceptor(Set.of("secret"));
    }
}
```

Register this factory so that it can be found at runtime by creating a `META-INF/services/io.smallrye.config.ConfigSourceInterceptorFactory` file that contains the fully qualified name of this factory class.

From this point forward, every lookup to the configuration name `secret` will throw a `SecurityException`.

> You can access Secret Keys by using this APIs `io.smallrye.config.SecretKeys#doUnlocked(java.lang.Runnable)` .

Reviewer: ```suggestion
Access the Secret Keys using the APIs `io.smallrye.config.SecretKeys#doUnlocked(java.lang.Runnable)` 
```
